internal/data_source_edgeFunction.go

```diff
@@ -0,0 +1,211 @@
+}
+
+func (d *EdgeFunctionDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, _ *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	d.client = req.ProviderData.(*apiClient)
+}
+
+func (d *EdgeFunctionDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_edge_function"
+}
+
+func (d *EdgeFunctionDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "Numeric identifier of the data source.",
+				Required:    true,
+			},
+			"schema_version": schema.Int64Attribute{
+				Description: "Schema Version.",
+				Computed:    true,
+			},
+			"results": schema.SingleNestedAttribute{
+				Computed: true,
+				Attributes: map[string]schema.Attribute{
+					"function_id": schema.Int64Attribute{
+						Description: "The function identifier.",
+						Computed:    true,
+					},
+					"name": schema.StringAttribute{
+						Description: "Name of the function.",
+						Computed:    true,
+					},
+					"language": schema.StringAttribute{
+						Description: "Language of the function.",
+						Computed:    true,
+					},
+					"code": schema.StringAttribute{
+						Description: "Code of the function.",
+						Computed:    true,
+					},
+					"json_args": schema.StringAttribute{
+						Computed:    true,
+						Description: "JSON arguments of the function.",
+					},
+					"function_to_run": schema.StringAttribute{
+						Description: "The function to run.",
+						Computed:    true,
+					},
+					"initiator_type": schema.StringAttribute{
+						Description: "Initiator type of the function.",
+						Computed:    true,
+					},
+					"active": schema.BoolAttribute{
+						Description: "Status of the function.",
+						Computed:    true,
+					},
+					"last_editor": schema.StringAttribute{
+						Description: "The last editor of the function.",
+						Computed:    true,
+					},
+					"modified": schema.StringAttribute{
+						Description: "Last modified timestamp of the function.",
+						Computed:    true,
+					},
+					"reference_count": schema.Int64Attribute{
+						Description: "The reference count of the function.",
+						Computed:    true,
+					},
+					"version": schema.StringAttribute{
+						Description: "Version of the function.",
+						Computed:    true,
+					},
+				},
+			},
+		},
+	}
+}
+
+func (d *EdgeFunctionDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var getEdgeFunctionId types.String
+	diags := req.Config.GetAttribute(ctx, path.Root("id"), &getEdgeFunctionId)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	edgeFunctionId, err := strconv.ParseUint(getEdgeFunctionId.ValueString(), 10, 16)
```

Comment:
To avoid long numbers problem, ParseUint should consider 32 bits instead of 16 bits